### PR TITLE
Redo portfolio and add line plots using Google Charts API

### PR DESCRIPTION
**Pull request by carolksun**
This PR changes the portfolio to a more interactive website using more advanced
features such as jQuery and AOS. It is based on this [tutorial](https://www.youtube.com/watch?v=T7PnWnTgusc) and was personalized 
to include a charts and comments section. This PR also uses the Google Charts 
API to plot raw and moving average sleep data from Carol's Fitbit.

### CODE REVIEW

**Comment by carolksun** [comment]
The website is running at [http://cksun-step-2020.appspot.com/](http://cksun-step-2020.appspot.com/)

File: portfolio/src/main/java/com/google/sps/servlets/SleepDataServlet.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                        arr[0],  
+                        arr[1],
+                        maSleepScore.get(counter),
+                        maDeepSleep.get(counter)
+                      };
+      counter++;
+      sleepData.put(date, data);
+    }
+  }
+}
```
Comment (razvanm) [inline portfolio/src/main/java/com/google/sps/servlets/SleepDataServlet.java]: Please add an end of line. Same in the other similar places.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -1,86 +1,253 @@
...
+  {packages:['corechart'], callback: 
+    function() { 
+      drawScoreChart();
+      drawMinChart();
+    }
+  });
+
+/**
+ * Creates line plot of raw data and corresponding 7-day moving average. To reduce
+ * code duplicity, the function is called for both the sleep score and the deep
```
Comment (razvanm) [inline portfolio/src/main/webapp/script.js]:
```suggestion
 * code duplication, the function is called for both the sleep score and the deep
```

File: portfolio/src/main/java/com/google/sps/servlets/SleepDataServlet.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  @Override
+  public void init() {
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream(FILENAME));
+    while (scanner.hasNextLine()) {
+        String line = scanner.nextLine();
+        String[] cells = line.split(",");
+
+        String timestamp = cells[0];
+        int tindex = timestamp.indexOf("T"); 
+        String date_string = timestamp.substring(0 , tindex);
```
Comment (razvanm) [inline portfolio/src/main/java/com/google/sps/servlets/SleepDataServlet.java:57]: This makes assumptions about the format of the timestamp. Can you explain what is going on and also show an example?

**Review by razvanm — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/SleepDataServlet.java (hunk 3 above)
Comment (carolksun) [reply portfolio/src/main/java/com/google/sps/servlets/SleepDataServlet.java:57]: I have the raw data in a csv file with the timestamp, sleep score, duration score, and deep sleep minutes. All the timestamps are of the form 2020-06-25T07:26:30Z, so I split it at "T" and take the first split. I added a comment in the java file explaining it.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -1,86 +1,253 @@
...
+    data.addColumn('number', 'Moving Average');
+    data.addColumn({type:'string', role:'annotation'});
+    data.addColumn({type:'string', role:'annotationText'}); 
+
+    /** Hard-coded specific dates to apply annotation text to certain portion of graph. */
+    Object.keys(sleep).forEach((day) => {
+      if (score) {
+        rawScore = sleep[day][0];
+        maScore = sleep[day][2];
+          if (day == "2020-05-30") {
```
Comment (razvanm) [inline portfolio/src/main/webapp/script.js]: This file seem to use both double quotes and single quotes for literal strings. The JavaScript style guide seem prefer single quotes.

Reference: https://google.github.io/styleguide/jsguide.html#features-strings-use-single-quotes

**Review by razvanm — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (carolksun) [reply portfolio/src/main/webapp/script.js]: fixed! For the strings that had the single quote character, I used the template strings as suggested by the style guide.

**Review by razvanm — APPROVED** [review]

**Comment by razvanm** [comment]
One more thing: note that the Markdown from the PR description will not show up nice in text form in the commit body.

**Pull request merged**